main/EventSystem.cpp

```diff
@@ -612,7 +613,7 @@ void CEventSystem::GetCurrentMeasurementStates()
 	m_windgustValuesByID.clear();
 	m_zwaveAlarmValuesByID.clear();
 
-	boost::shared_lock<boost::shared_mutex> devicestatesMutexLock(m_devicestatesMutex);
+	std::shared_lock<boost::shared_mutex> devicestatesMutexLock(m_devicestatesMutex);
```

Comment:
Is it maybe an idea to also use std::shared_mutex ... if it needs to be shared, maybe std::mutex will work as well ?

Reply:
std::shared_mutex is C++17

Reply:
True, but does it need to be a shared mutex or is a normal mutex sufficient ?

Reply:
I don't know. I prefer not to edit the code.

Reply:
Is there a serious reason for not switching to C++17 anyway?